client/public/global-config.js

```diff
@@ -0,0 +1,5 @@
+globalConfig = {
```

Comment:
Don't do this :(
You are introducing a global variable in a very bad way. Can you avoid doing so?

Reply:
This is a pattern for being able to inject deployment-time configuration information into a client-side application.

React / node has ways to inject compile time configuration information, e.g. in a development vs production build, for example the build number, through environment config overrides. But this is bundled into the compiled deployment artefacts.

If you want to deploy the same artefact to multiple environments, e.g. Staging and Production, and have different values for those configuration settings, e.g. a different instrumentation key for each environment, then you need some way to change that setting.

There are several ways to do this.  Having a search and replace into the already bundled code, for some special keyword, is one way, but not easily done manually.  If you know the limited environments you will be using, then a big "if" statement that checks something like the URL can be used.  Another way is to wait until after the app has started up, and then make a web request call to the server to get configuration settings (servers have much better solutions for configuration, e.g. appsettings.json file for .NET apps).

Yes, this is technically using a global javascript variable, but it is using it as a means to inject an environment specific config (which should be used as static values); the global-config.js file is deployed independent of the main app bundle, and so easy to do an environment-specific replace on.

Goto is also a bad thing, but ultimately every bit of code is going to be compiled down to machine language instructions littered with JUMP commands; the entire TypeScript transpiling to Javascript also uses/misuses low level language features to achieve higher level concerns.

If you want to suggest a better way to inject deployment-environment specific configuration, feel free to provide a suggested code change. (As mentioned, there are many ways to do this).